File: hrms-frontend/src/services/complianceService.js
// src/services/complianceService.js
import api from '../api/axiosConfig';

const complianceService = {
  // Records
  getAllRecords: () => api.get('/compliance/records'),
  getRecord: (id) => api.get(`/compliance/records/${id}`),
  getRecordsByType: (type) => api.get(`/compliance/records/type/${type}`),
  getRecordsByPeriod: (year, month) =>
    api.get(`/compliance/records/period?year=${year}${month ? `&month=${month}` : ''}`),
  createRecord: (data) => api.post('/compliance/records', data),
  updateRecord: (id, data) => api.put(`/compliance/records/${id}`, data),
  deleteRecord: (id) => api.delete(`/compliance/records/${id}`),
  markFiled: (id, acknowledgmentNumber) =>
    api.patch(`/compliance/records/${id}/file`, { acknowledgmentNumber }),
  markPaid: (id, challanNumber) =>
    api.patch(`/compliance/records/${id}/pay`, { challanNumber }),

  // Events
  getAllEvents: () => api.get('/compliance/events'),
  getUpcoming: (days = 30) =>
    api.get(`/compliance/events/upcoming?days=${days}`),
  getOverdue: () => api.get('/compliance/events/overdue'),
  createEvent: (data) => api.post('/compliance/events', data),
  updateEvent: (id, data) => api.put(`/compliance/events/${id}`, data),
  completeEvent: (id, notes) =>
    api.patch(`/compliance/events/${id}/complete`, { notes }),
  deleteEvent: (id) => api.delete(`/compliance/events/${id}`),

  getDashboard: () => api.get('/compliance/dashboard'),
  getReport: (year) => api.get(`/compliance/report?year=${year}`),
};

export default complianceService;

File: hrms-backend/src/main/java/com/hrms/service/ComplianceService.java

//service/ComplianceService.java
package com.hrms.service;

import com.hrms.model.ComplianceEvent;
import com.hrms.model.StatutoryRecord;

import java.time.LocalDate;
import java.util.List;
import java.util.Map;

public interface ComplianceService {

 // Statutory Records
 StatutoryRecord createRecord(StatutoryRecord record);
 StatutoryRecord updateRecord(Long id, StatutoryRecord record);
 StatutoryRecord getRecord(Long id);
 List<StatutoryRecord> getAllRecords();
 List<StatutoryRecord> getRecordsByType(String type);
 List<StatutoryRecord> getRecordsByPeriod(Integer year, Integer month);
 List<StatutoryRecord> getRecordsByStatus(String status);
 void deleteRecord(Long id);

 StatutoryRecord markFiled(Long id, String acknowledgmentNumber);
 StatutoryRecord markPaid(Long id, String challanNumber);

 // Compliance Events
 ComplianceEvent createEvent(ComplianceEvent event);
 ComplianceEvent updateEvent(Long id, ComplianceEvent event);
 ComplianceEvent getEvent(Long id);
 List<ComplianceEvent> getAllEvents();
 List<ComplianceEvent> getUpcomingEvents(int days);
 List<ComplianceEvent> getOverdueEvents();
 ComplianceEvent completeEvent(Long id, String notes);
 void deleteEvent(Long id);

 // Analytics
 Map<String, Object> getDashboardStats();
 Map<String, Object> getComplianceReport(Integer year);
}